LMS/Studio: Sass Build Files to Reduce Compilation Redundancy

talbs opened the pull request:
This work moves any common Sass imports shared across LTR/RTL Sass compile files into one ``_build-*.scss`` partial for easier maintenance. 

Note: There are some elements that are still shared in the LMS that needed to be outside of this build (for Stanford theming override purposes).

talbs commented:
test this please

talbs commented:
NOTE: This work is based off of https://github.com/edx/edx-platform/pull/7866.

talbs commented on lms/static/sass/application-rtl.scss.mako:
```diff
@@ -28,72 +28,7 @@
   @import '${env.get('THEME_NAME')}';
 % endif
 
-@import 'base/base';
```
NOTE: sadly, we cannot abstract more out of this compile list due to theming configuration still being controlled through Mako logic. The rest need to stay or I can create more ``_build-*.scss`` files, but that seems less clear and more messy.

frrrances replied:
Yup, agreed. Once we move away from mako variables, we can add in the rest of the import to the build file.

talbs commented on lms/static/sass/application.scss.mako:
```diff
@@ -28,72 +28,7 @@
   @import '${env.get('THEME_NAME')}';
 % endif
 
-@import 'base/base';
```
NOTE: sadly, we cannot abstract more out of this compile list due to theming configuration still being controlled through Mako logic. The rest need to stay or I can create more ``_build-*.scss`` files, but that seems less clear and more messy.

talbs commented on lms/static/sass/course-rtl.scss.mako:
```diff
@@ -19,59 +19,9 @@
   @import '${env.get('THEME_NAME')}';
 % endif
 
-@import 'base/base';
```
NOTE: sadly, we cannot abstract more out of this compile list due to theming configuration still being controlled through Mako logic. The rest need to stay or I can create more ``_build-*.scss`` files, but that seems less clear and more messy.

talbs commented on lms/static/sass/course.scss.mako:
```diff
@@ -19,66 +19,9 @@
   @import '${env.get('THEME_NAME')}';
 % endif
 
-@import 'base/base';
```
NOTE: sadly, we cannot abstract more out of this compile list due to theming configuration still being controlled through Mako logic. The rest need to stay or I can create more ``_build-*.scss`` files, but that seems less clear and more messy.

talbs commented:
@frrrances and @clrux, this is a follow-up to https://github.com/edx/edx-platform/pull/8000 to help clean up and maintain our compile lists in a more central way. Mind giving this a review?

clrux commented:
Did you test this in IE9?!

frrrances commented on lms/static/sass/application.scss.mako:
```diff
@@ -28,72 +28,7 @@
+@import 'build-lms'; // shared app style assets/rendering
 
 ## Import styles for search
 % if env["FEATURES"].get("ENABLE_DASHBOARD_SEARCH", False):
```
Hmm... Hadn't seen this before. I know it's not your commit, but it might be worth digging in at some point to see if its necessary to exclude this Sass import. Usually it doesn't hurt much to include it, and it's one less non-Sass/CSS thing in our import...

talbs replied:
Hmm, good point. I'll take al look through the partial and make sure rules are scoped to the feature's attributes/expected DOM (just in case there are any layout or wacky resets happening). 

Also, @marcotuts may know more than both of us on this partial's contents/scope.

marcotuts replied:
The benefit here was mostly to try and keep the styles file smaller for instances that aren't using search-based features. (That I think was the original intent, from back when courseware search added this.) That being said, we're increasingly relying on features that use search styles (bookmarks for example), and will be soon enabling this by default. In fact, an existing PR currently nearing merge removes this if statement.

talbs replied:
Cool, I'll take a look through the rules and make sure they're scoped before removing this. Thanks!

frrrances commented:
Nice work, @talbs! One question about the new if statement related to search, but not a blocker. :+1:

marcotuts commented:
Added a comment re: the search styles. A current in-progress PR removes this conditional wrapper, so it seems safe to remove here as well, or rely on that upcoming PR to test / deal with the removal of the conditional. Excited to see this land!

talbs commented:
Okay, with https://github.com/edx/edx-platform/commit/220b8d1cdb3ca37ce15b4947c8d72bdc901cb3bc I've removed the search mako conditional logic and simplified the ``_build-*.scss`` files even more given that. I'll wait till tests pass and all is well.

Thanks for the review.

Once this is done, simple nice steps of cleaning things up can happen. Next stop - consistent application Sass/CSS file names!

frrrances commented:
wahooo! :stars:

talbs commented:
test this please
